### PR TITLE
Fix plain http connection not proxied

### PR DESCRIPTION
**Pull request by alesiong**
Set `Connection` for `ProxyStream::Regular` to proxied, otherwise hyper will not append host before http path

### CODE REVIEW

**Comment by alesiong** [comment]
@howardmoore 
Thanks for your review. I added a `ProxyStream::NoProxy` variant for the non-proxied plain http connection.

File: src/lib.rs [hunk 1]
```diff
@@ -429,7 +429,7 @@ where
                             #[cfg(not(any(feature = "tls", feature = "rustls")))]
                             Some(_) => panic!("hyper-proxy was not built with TLS support"),
 
-                            None => Ok(ProxyStream::Regular(tunnel_stream)),
+                            None => Ok(ProxyStream::NoProxy(tunnel_stream)),
```
Comment (howardmoore) [inline src/lib.rs:432]: I think this should still be ProxyStream::Regular, shouldn't it? You've still matched a proxy here, you just don't have any TLS support enabled.

Comment (alesiong) [reply src/lib.rs]: fixed

Comment (howardmoore) [reply src/lib.rs]: Looks good.

**Comment by tafia** [comment]
Thanks!

**Comment by tafia** [comment]
published v0.7.0

**Pull request merged**